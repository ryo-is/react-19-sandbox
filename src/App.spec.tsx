import { describe, expect, it } from 'vitest';
import { render, screen } from '@testing-library/react';
import { userEvent } from '@testing-library/user-event';

import App from './App';

describe('App', () => {
  it('snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment).toMatchSnapshot();
  });

  it('display h1', () => {
    render(<App />);
    const headerElement = screen.getByText('Vite + React');
    expect(headerElement).toBeInTheDocument();
  });

  it('count up', async () => {
    render(<App />);

    const afterClickButton = screen.getByRole('button', { name: 'count is 0' });
    await userEvent.click(afterClickButton);

    const beforeClickButton = screen.getByRole('button', {
      name: 'count is 1',
    });
    expect(beforeClickButton).toBeInTheDocument();
    expect(afterClickButton).toHaveTextContent('count is 1');
  });
});
