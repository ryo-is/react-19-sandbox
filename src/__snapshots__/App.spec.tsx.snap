// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > snapshot 1`] = `[Function]`;
